src/Http/Http.Results/src/Results.cs

```diff
@@ -690,6 +691,29 @@ public static IResult UnprocessableEntity(object? error = null)
     public static IResult UnprocessableEntity<TValue>(TValue? error)
         => error is null ? TypedResults.UnprocessableEntity() : TypedResults.UnprocessableEntity(error);
 
+    /// <summary>
+    /// Produces a <see cref="StatusCodes.Status500InternalServerError"/> response.
+    /// </summary>
+    /// <returns>The created <see cref="IResult"/> for the response.</returns>
+    public static IResult InternalServerError()
+        => InternalServerError<object>(null);
+
+    /// <summary>
+    /// Produces a <see cref="StatusCodes.Status500InternalServerError"/> response.
+    /// </summary>
+    /// <param name="error">An error object to be included in the HTTP response body.</param>
+    /// <returns>The created <see cref="IResult"/> for the response.</returns>
+    public static IResult InternalServerError<TValue>(TValue? error)
+        => error is null ? TypedResults.InternalServerError() : TypedResults.InternalServerError(error);
+
+    /// <summary>
+    /// Produces a <see cref="StatusCodes.Status500InternalServerError"/> response.
+    /// </summary>
+    /// <param name="exception">The exception to be thrown.</param>
+    /// <returns>The created <see cref="IResult"/> for the response.</returns>
+    public static IResult InternalServerError(ExceptionDispatchInfo exception)
```

Comment:
We just [approved the API](https://github.com/dotnet/aspnetcore/issues/53073#issuecomment-1961936888) with everything except these `ExceptionDispatchInfo` overloads. I know it was our suggestion in the first place, so sorry to give you the runaround.

Thanks for sticking with this as we went through the API review process.

Reply:
Yes. Thank you!

Reply:
Done!